### DISCUSSION TITLE
How to find data from joined table?

### DISCUSSION THREAD

**Discussion by hiscaler** [cycle/orm#444] [Q&A]
```php
#[Entity]
class User {

protected string $name;

}

#[JoinedTable]
class Admin extends User
{

protected string $age;
}

$orm->getRepository(User::class)->findAll();
```

How do can access Admin.age attribute?

**Comment by hiscaler** [comment]
First use instanceof check class, and use it, like this
```php
 if ($user instanceof Admin) {
// todo
}
```

**Comment by hiscaler** [comment]
Other a question, how to insert a record to admin table, if user table has one record, and admin table is no record. like this:


user table

id username
1  admin

admin table

id remark
no records

no records in admin table

i will insert a record to admin table
id remark
1  this is a administrator

**Reply by roxblnfk** [reply]
Do I understand correctly that you want to convert a User into an Admin?

**Reply by hiscaler** [reply]
yes

**Reply by roxblnfk** [reply]
I'm not sure that's possible using ORM only without recreating the entity with a new role

**Reply by wolfy-j** [reply]
This is not really possible to do automatically since ORM does not know how to fill remaining properties of the Admin.

@roxblnfk what if we will simply replace the entity in Heap? We can expose this function and in this case it will be possible to replace the binding, but any entities that relate to older user might be broken...

**Reply by roxblnfk** [reply]
In any case, this is a new mechanism that requires careful consideration.

**Comment by roxblnfk** [comment]
> How do can access Admin.age attribute?

If you correctly set the Entity and Column attributes, the entity property will be automatically filled. You can either make the property public or add a getter.

**Reply by hiscaler** [reply]
new Admin()?
but will generate a new User record.
Now condition is User is exists, i will bind a Admin to exists User
